### PR TITLE
Feature/add example dir

### PR DESCRIPTION
**Pull request by kcoleman731**
*Description of changes:*

This is a prototype/proposal PR. I've tried to organize the example code in a way that is easier to read/navigate from a consumer point of view. I've moved the stacks from `bin/main.ts` into dedicated example dirs and files in the `examples` directory. Lets discuss if we want to keep the example code in this project or move to the `ssp-patterns` repo. Perhaps we keep a single example for testing, but move the rest?

I also moved created a `BottlerocketClusterProvider` class in the `cluster-provider` dir. Not sure if that is what we ant @shapirov103?

Cleaned up imports and exports as well. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: examples/fargate-stack/index.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// Team implementations
+import * as team from '../teams'
+
+export default class MultiTeamStack extends cdk.Stack {
+    constructor(app: cdk.App, id: string, props?: cdk.StackProps) {
+        super(app, id, props);
+
+        // Teams for the cluster.
+        const teams: Array<ssp.Team> = [
+            new team.TeamPlatform,
```
[the remark this answers is not in the document]
Comment (kcoleman731) [reply examples/fargate-stack/index.ts:14]: wrong name. This is the Fargate stack

File: examples/multi-team-stack/index.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import * as cdk from '@aws-cdk/core';
+
+// SSP Lib
+import * as ssp from '../../lib'
+
+// Team implementations
+import * as team from '../teams'
+
+export default class MultiTeamStack {
+    constructor(app: cdk.App, id: string, props?: cdk.StackProps) {
```
Comment (shapirov103) [inline examples/multi-team-stack/index.ts:10]: Linting passed? I see lint warnings on it.

File: examples/teams/team-platform/index.ts [hunk 3]
```diff
@@ -1,11 +1,12 @@
...
-import { PlatformTeam } from "..";
+
+import { PlatformTeam } from '../../../lib/teams';
 
 export class TeamPlatform extends PlatformTeam {
     constructor() {
         super({
             name: "platform",
-            users: [new ArnPrincipal("arn:aws:iam::929819487611:user/superadmin")]
+            users: [new ArnPrincipal("arn:aws:iam::115717706081:user/superadmin")]
```
Comment (shapirov103) [inline examples/teams/team-platform/index.ts:9]: Why account change? If it is still a personal account (isengard) then I see no reason to change. One approach is to parameterize the account and make it part of the cdk context (see EC2 cluster provider options). Could be obtained as `anyStructLikeStackOrCluster.node.tryGetContext()`.

File: lib/index.ts.orig [hunk 4]
```diff
@@ -0,0 +1,15 @@
+<<<<<<< HEAD
```
Comment (shapirov103) [inline lib/index.ts.orig:1]: Is this file needed? Can we drop it?

File: Makefile [hunk 5]
```diff
@@ -17,6 +17,9 @@ lint:
 build:
 	rm -rf dist && $(TSC)
 
+deploy:
```
[the remark this answers is not in the document]
Comment (shapirov103) [reply Makefile:20]: Why do we need this command? Is it consistency only or am i missing something?

**Review by shapirov103 — COMMENTED** [review]
A few minor comments. Seems ready to be moved out. It would be nice to get the official aws npm for them though.

**Review by parkand1 — APPROVED** [review]
LGTM
